src/api/hooks/video.ts

```diff
@@ -109,3 +112,50 @@ export const useMostViewedVideos = (variables?: GetMostViewedVideosQueryVariable
     ...rest,
   }
 }
+
+type MostViewedVideosAllTimeOpts = QueryHookOptions<GetMostViewedVideosAllTimeQuery>
+export const useMostViewedVideosAllTimeIds = (
+  variables?: GetMostViewedVideosAllTimeQueryVariables,
+  opts?: MostViewedVideosAllTimeOpts
+) => {
+  const { data, ...rest } = useGetMostViewedVideosAllTimeQuery({ ...opts, variables })
+  return {
+    mostViewedVideosAllTime: data?.mostViewedVideosAllTime,
+    ...rest,
+  }
+}
+
+export const useMostViewedVideosAllTime = (
+  variables?: GetMostViewedVideosAllTimeQueryVariables,
+  opts?: MostViewedVideosAllTimeOpts
+) => {
+  const { mostViewedVideosAllTime } = useMostViewedVideosAllTimeIds(variables, opts)
+
+  const mostViewedVideosAllTimeIds = useMemo(() => {
+    if (mostViewedVideosAllTime) {
+      return mostViewedVideosAllTime.map((item) => item.id)
+    }
+    return null
+  }, [mostViewedVideosAllTime])
+
+  const { videos, ...rest } = useVideos(
+    {
+      where: {
+        id_in: mostViewedVideosAllTimeIds,
+      },
+    },
+    { skip: !mostViewedVideosAllTimeIds }
+  )
+
+  const sortedVideos = useMemo(() => {
+    if (videos) {
+      return [...videos].sort((a, b) => (b.views && a.views ? b.views - a.views : 0))
```

Comment:
If I'm not mistaken, because of `b.views && a.views` condition, if one of the items doesn't have views but the other does, the sort may be incorrect right?